code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/load/AbstractUnifier.kt

```
@@ -0,0 +1,74 @@
+package org.jetbrains.research.code.submissions.clustering.load
+
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.jetbrains.python.PythonFileType
+import org.jetbrains.research.code.submissions.clustering.model.Submission
+import org.jetbrains.research.code.submissions.clustering.util.addFileToProject
+import org.jetbrains.research.code.submissions.clustering.util.getTmpProjectDir
+import org.jetbrains.research.ml.ast.transformations.Transformation
+
+/**
+ * Enum class for possible code submissions' languages.
+ * @property extension corresponding file extension
+ */
+enum class Language(val extension: String) {
+    PYTHON(PythonFileType.INSTANCE.defaultExtension),
+    ;
+}
+
+/**
+ * Abstract unifier producing unifying transformations over code submissions.
+ */
+abstract class AbstractUnifier {
+    abstract val project: Project
+    abstract val psiManager: PsiManager
+    abstract val language: Language
+    abstract val transformations: List<Transformation>
+
+    private fun String.createPsiFile(id: Int): PsiFile = ApplicationManager.getApplication().runWriteAction<PsiFile> {
+        val basePath = getTmpProjectDir(toCreateFolder = false)
+        val fileName = "dummy$id.${language.extension}"
+        val file = addFileToProject(basePath, fileName, fileContext = this)
+        val virtualFile = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file)
```

Please, throw an error if the file was not found